### DISCUSSION TITLE
Is there a way to get `explicit_search_params` for the current route?

### DISCUSSION THREAD

**Discussion by ranjan-purbey** [jycouet/kitql#578] [Q&A] [answered]
Can something like the following be used in the load function?
```js
  const {page, limit} = getRouteSp(request)
```
or
```js
  const page = request.searchParams.get( explicit_sp.page )
```

**Comment by jycouet** [comment, marked as answer]
I believe that if you have your config as a separate object, yes. 👍
You have a specific usecase in mind? 

With https://github.com/sveltejs/kit/issues/11386 point 3️⃣ want's to address this.

**Comment by ranjan-purbey** [comment]
Thanks @jycouet ! Breaking the config out into a separate object would solve the problem to certain extent. My use case is simple - to have the compiler catch wrong keys in `searchParams.get` calls.

This may be out of scope of the plugin, but augmenting the `RequestEvent` type for the route with a `searchParams` property similar to `params` could be helpful. It could derive the allowed keys from either a generic type parameter on the load function or an exported keyword type from the `+page.ts`, as you suggested in your proposal.

**Reply by jycouet** [reply]
Maybe I didn't understand correctly your answer, but you can already do a separate config, and make use of it where you want 👍

For the proposal, it's what I would like to have, native in sveltekit, in event! Let's push this forward 🚀

**Reply by ranjan-purbey** [reply]
I was just agreeing with you on your suggestion to break out the config. I too would love to have this baked into the framework. Happy to help :)

**Reply by ranjan-purbey** [reply]
Btw this plugin is awesome!! Thanks for your work! I just found it yesterday through your tweet about the above proposal. And today our project is already using it. Migrating to it was such a pleasure :heart:

**Reply by jycouet** [reply]
Love to read this, thx
